File: bindings/dotnet/src/Zlink/Runtime/Sockets/XSubSocket.cs
// SPDX-License-Identifier: MPL-2.0

namespace Systems.Zlink;

public sealed class XSubSocket : SubscriberSocketBase, IXSubSocket
{
    public new SubSocketOptions Options { get; }

    public XSubSocket(Context context)
        : base(context, SocketType.XSub)
    {
        Options = new SubSocketOptions(this);
    }
}

File: bindings/node/src/zlink/runtime/sockets/xsub_socket.ts
// SPDX-License-Identifier: MPL-2.0

import { SubSocketOptions } from './socket_options';
import {
  SubscriberSocket,
} from './socket_operations';
import {
  NativeSocketType,
  type Context,
} from './socket_runtime_support';

export class XSubSocket extends SubscriberSocket {
  readonly options: SubSocketOptions;
  constructor(ctx: Context) { super(ctx, NativeSocketType.XSUB); this.options = SubSocketOptions.create(this); }
}
